tests/test_plan.py

```diff
@@ -934,19 +890,27 @@ def test_plan_link_menuinst(self):
         assert expected_plan[2] == conda_plan[2]
 
 
-def generate_mocked_package(preferred_env, name, schannel, version):
-    mock_package = namedtuple("Package", ["preferred_env", "name", "schannel", "version", "fn"])
-    return mock_package(preferred_env=preferred_env, name=name, schannel=schannel,
-                       version=version, fn=name)
-
-
-def generate_mocked_resolve(matched_pkgs, index, install=None):
+def generate_mocked_resolve(pkgs, install=None):
```

Comment:
This now builds `Dist` and mocked `IndexRecord` objects, puts in them in an index dictionary, *and* populates an array of `get_pkgs` matches.